### PR TITLE
make put operation also has required body parameters

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Regeneration: https://github.com/Azure/azure-sdk-for-net/pull/29632

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/Azure.Management.Storage/Generated/RestOperations/BlobContainersRestOperations.cs [hunk 1]
```diff
@@ -693,13 +693,10 @@ internal HttpMessage CreateCreateOrUpdateImmutabilityPolicyRequest(string subscr
...
             }
             request.Headers.Add("Accept", "application/json");
-            if (data != null)
-            {
-                request.Headers.Add("Content-Type", "application/json");
-                var content = new Utf8JsonRequestContent();
-                content.JsonWriter.WriteObjectValue(data);
-                request.Content = content;
-            }
+            request.Headers.Add("Content-Type", "application/json");
```
Comment (ArthurMa1978) [inline samples/Azure.Management.Storage/Generated/RestOperations/BlobContainersRestOperations.cs:696]: throw exception if data is null?

File: src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs [hunk 2]
```diff
@@ -43,7 +43,7 @@ internal static void Update()
         private IEnumerable<Schema> allGeneralSchemas;
         private IEnumerable<OperationGroup> allOperationGroups;
         private IReadOnlyDictionary<string, string> allFormatByNameRules;
-        private Dictionary<Guid, string> schemaCache = new Dictionary<Guid, string>();
+        private Dictionary<Schema, string> schemaCache = new();
```
Comment (ArthurMa1978) [inline src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs:46]: Not sure here the `Schema` as key would drop down the perf

**Review by ArthurMa1978 — APPROVED** [review]

File: src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs (hunk 2 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs:46]: Absolutely no. We are doing that a lot (in MgmtOutputLibrary). These factors impact performance very minorly

File: samples/Azure.Management.Storage/Generated/RestOperations/BlobContainersRestOperations.cs (hunk 1 above)
Comment (ArcturusZhang) [reply samples/Azure.Management.Storage/Generated/RestOperations/BlobContainersRestOperations.cs:696]: We assert the data is not null in the corresponding client method. Therefore we no longer need to assert again

File: src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs (hunk 2 above)
Comment (Yao725) [inline src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs:46]: Just for curiosity, why is this change needed?

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Mgmt/Decorator/Transformer/SchemaFormatByNameTransformer.cs:46]: 1. If we have to introduce an Id in schema, we should not do that in the CodeModel.cs file. We should put it in CodeModelPartial.
2. This has no point since the instance itself works as an unique Id of a schema object. We are doing that a lot in MgmtOutputLibrary, quite a few dictionaries are using the objects from code model as keys (such as Schema, Operation, etc)
